### DISCUSSION TITLE
reload=True in uvicorn.run() to work with app instance

### DISCUSSION THREAD

**Discussion by denyszhak** [Kludex/uvicorn#2135] [Q&A] [answered]
I see there is a constraint in the code to enable auto-reload (reload=True) only when the app argument to uvicorn.run() is an app path string, but it doesn't work if that's an instance.

The question why not allow to work reload when passing the app as an instance? In my use case, I don't want to initialize settings I pass to the app globally.

Any workaround also work, txs.

```
def run_app() -> None:
    settings = Settings()
    app = provide_app(settings)

    uvicorn.run(
        app=app,
        workers=settings.NUMBER_OF_WORKERS,
        host=settings.SERVICE_HOST,
        port=settings.SERVICE_PORT,
    )

if __name__ == "__main__":
    run_app()
```

**Comment by Kludex** [comment, marked as answer]
Because we run another process on reload, so we pass the path of the app to be loaded on another process.

**Reply by Kludex** [reply]
We could extract the app location, I think? If you submit a PR, and the code is cute, I can accept it.
